src/main/java/com/ngs/stash/externalhooks/hook/ExternalMergeCheckHook.java

```diff
@@ -0,0 +1,222 @@
+        Repository repo = pr.getToRef().getRepository();
+
+        // compat with Stash < 3.2.0
+        String repoPath = this.properties.getRepositoryDir(repo).getAbsolutePath();
+
+        Settings settings = context.getSettings();
+        List<String> exe = new LinkedList<String>();
+        exe.add(this.getExecutable(
+            settings.getString("exe"),
+            settings.getBoolean("safe_path", false)).getPath());
+
+        if (settings.getString("params") != null) {
+            for (String arg : settings.getString("params").split("\r\n")) {
+                exe.add(arg);
+            }
+        }
+
+        StashUser currentUser = authCtx.getCurrentUser();
+        ProcessBuilder pb = new ProcessBuilder(exe);
+
+        Map<String, String> env = pb.environment();
+        env.put("STASH_USER_NAME", currentUser.getName());
+        if (currentUser.getEmailAddress() != null) {
+            env.put("STASH_USER_EMAIL", currentUser.getEmailAddress());
+        } else {
+            log.error("Can't get user email address. getEmailAddress() call returns null");
+        }
+        env.put("STASH_REPO_NAME", repo.getName());
+
+        boolean isAdmin = permissions.hasRepositoryPermission(
+            currentUser, repo, Permission.REPO_ADMIN);
+        env.put("STASH_IS_ADMIN", String.valueOf(isAdmin));
+
+        RepositoryCloneLinksRequest.Builder cloneLinksRequestBuilder =
+            new RepositoryCloneLinksRequest.Builder();
+
+        cloneLinksRequestBuilder.repository(repo);
+
+        RepositoryCloneLinksRequest cloneLinksRequest =
+            cloneLinksRequestBuilder.build();
+
+        Set<NamedLink> cloneLinks = this.repoService.getCloneLinks(
+            cloneLinksRequest
+        );
+
+        for (NamedLink link : cloneLinks) {
+            env.put(
+                "STASH_REPO_CLONE_" + link.getName().toUpperCase(),
+                link.getHref()
+            );
+        }
+
+        env.put(
+            "STASH_BASE_URL",
+            this.properties.getBaseUrl().toString()
+        );
+
+        env.put("STASH_PROJECT_NAME", repo.getProject().getName());
+        env.put("STASH_PROJECT_KEY", repo.getProject().getKey());
+
+        String summaryMsg = "Merge request failed";
+
+        pb.directory(new File(repoPath));
+        pb.redirectErrorStream(true);
+        try {
+            Process process = pb.start();
+            InputStreamReader input = new InputStreamReader(
+                process.getInputStream(), "UTF-8");
+            OutputStream output = process.getOutputStream();
+
+            output.write((
+                          pr.getFromRef().getRepository().getSlug() + " " +
+                          pr.getFromRef().getLatestChangeset() + " " +
+                          pr.getToRef().getRepository().getSlug() + " " +
+                          pr.getToRef().getLatestChangeset() + "\n"
+                          ).getBytes("UTF-8"));
+            output.close();
```

Comment:
> I don't know of any way to get the branch name. If someone can point me to it I can pass it along. Or just add to this pull request. :)

`pr.getToRef().getDisplayId()`, like I wrote before.

> For a merge check, there is a lot more information to consider.

Can you give an example? Because I cannot think of any. In my world, a pre-receive hook checks whether to allow or not a ref (branch) to be moved from A to B. A merge/PR hook checks whether to allow a ref (target branch) to be moved from A to B', where B' is actually the merge of B into the HEAD of the target branch. Same thing, in a nutshell. Given that, in the best of worlds, I would like to use the same hook for pre-receive and PR, by default. But that's irrelevant for this PR.

> Splitting input between stdio and environment variables seem arbitrary. I'm thinking the information should all be passed in the same way.

Not sure what to make of this. You're now arguing that there should be no environment variables i.e. STASH_PROJECT_NAME should be the n-th parameter passed via STDIN ?

I'm sure though that the trick for me to understand better is to hear you exemplify what more can make your PR git hook pass or fail. For me, information like PR id, author, source branch, version, comments, etc cannot influence the outcome.